packages/astro/src/content/runtime.ts

```diff
@@ -414,13 +414,20 @@ async function updateImageReferencesInBody(html: string, fileName: string) {
 	for (const [_full, imagePath] of html.matchAll(CONTENT_LAYER_IMAGE_REGEX)) {
 		try {
 			const decodedImagePath = JSON.parse(imagePath.replaceAll('&#x22;', '"'));
-			const id = imageSrcToImportId(decodedImagePath.src, fileName);
 
-			const imported = imageAssetMap.get(id);
-			if (!id || imageObjects.has(id) || !imported) {
-				continue;
+			let image: GetImageResult;
+			if (validURL(decodedImagePath.src)) {
+				// Remote image, pass through without resolving import
+				image = await getImage(decodedImagePath);
```

Comment:
Shouldn't we validate if the remote image we are about to download is part of the allowed domains? https://docs.astro.build/en/guides/images/#authorizing-remote-images